app/models/concerns/orchestration/tftp.rb

```diff
@@ -91,7 +107,10 @@ def setTFTP(kind)
   # +returns+ : Boolean true on success
   def delTFTP(kind)
     logger.info "Delete the TFTP configuration for #{host.name}"
-    tftp.delete kind, mac
+    results = []
+    results << tftp.delete(kind, mac) if tftp?
+    results << tftp6.delete(kind, mac) if tftp6?
```

Comment:
Same comment as https://github.com/theforeman/foreman/pull/3626/files#r77612779 in all of these methods - get an array of `[tftp, tftp6]` de-duplicated and then iterate (well, call `.all? { |p| p.delete(..) }` I suppose).